devops/scripts/benchmarks/utils/unitrace.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (C) 2024-2025 Intel Corporation
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+import os
+import shutil
+import re
+
+from options import options
+from utils.utils import run, git_clone
+from utils.oneapi import get_oneapi
+
+
+def extract_save_name_and_timestamp(dirname):
+    """
+    Extracts (save_name, timestamp) from a directory name of the form {save_name}_{timestamp},
+    where timestamp is always 15 characters: YYYYMMDD_HHMMSS.
+    save_name may contain underscores.
+    """
+    m = re.match(r"(.+)_(\d{8}_\d{6})$", dirname)
+    if m:
+        return m.group(1), m.group(2)
+    return None, None
+
+
+def prune_unitrace_dirs(base_dir, FILECNT=10):
```

Comment:
This is used only in this module, please rename this method with an underscore prefix to make it clearer that it's for an internal use of this module's methods